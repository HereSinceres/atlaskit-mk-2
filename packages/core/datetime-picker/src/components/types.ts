export type Appearance = 'default' | 'subtle';
export type Spacing = 'compact' | 'default';

// Similiar to T = {...A, ...B}
export type Combine<First, Second> =
  // 1. Remove all overlapping types from First
  // 2. Add properties from Second
  Omit<First, keyof Second> & Second;
