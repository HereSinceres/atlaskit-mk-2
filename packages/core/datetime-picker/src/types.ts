export type Appearance = 'default' | 'subtle';
export type Spacing = 'compact' | 'default';

export type SelectProps = any;

export type PassThrough = React.AllHTMLAttributes<HTMLElement>;
