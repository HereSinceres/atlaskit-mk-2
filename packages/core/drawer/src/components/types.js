// @flow
import type { ComponentType, Node } from 'react';
import type { WithAnalyticsEventsProps } from '@atlaskit/analytics-next';

export type DrawerWidth = 'narrow' | 'medium' | 'wide' | 'extended' | 'full';

export type BaseProps = {
  /** The content of the drawer */
  children: Node,
  /** Icon to be rendered in your drawer as a component, if available */
  icon?: ComponentType<*>,
  /** Available drawer sizes */
  width: DrawerWidth,
};

export type DrawerPrimitiveProps = BaseProps & {
  in: boolean,
  onClose?: (SyntheticMouseEvent<*>) => void,
  onCloseComplete?: (node: HTMLElement) => void,
  shouldUnmountOnExit?: boolean,
};

export type DrawerProps = BaseProps & {
  ...WithAnalyticsEventsProps,
  /**
      Callback function to be called when the drawer will be closed.
    */
  onClose?: (
    SyntheticMouseEvent<*> | SyntheticKeyboardEvent<*>,
    analyticsEvent: any,
  ) => void,
  /** A callback function that will be called when the drawer has finished its close transition. */
  onCloseComplete?: (node: HTMLElement) => void,
  /**
      Callback function that will be called when the drawer is displayed and `keydown` event is triggered.
    */
  onKeyDown?: (SyntheticKeyboardEvent<*>) => void,
  /** Controls if the drawer is open or close */
  isOpen: boolean,
  /** Boolean that controls if drawer should be retained/discarded */
  shouldUnmountOnExit?: boolean,
  /**
    Boolean indicating whether to focus on the first tabbable element inside the drawer.
  */
  autoFocusFirstElem?: boolean,
  /**
   *     Boolean indicating if the focus lock is active or not.
   */

  isFocusLockEnabled?: boolean,
  /**
    Whether to return the focus to the previous active element on closing the drawer
  */
  shouldReturnFoucs?: boolean,
};

type DnDType = {
  draggableProps: {
    style: ?Object,
    'data-react-beautiful-dnd-draggable': string,
  },
  dragHandleProps: ?Object,
  innerRef: Function,
  placeholder?: Node,
};

export type ItemProps = {
  /** Whether the Item should attempt to gain browser focus when mounted */
  autoFocus?: boolean,
  /** Main content to be shown inside the item. */
  children?: Node,
  /** Secondary text to be shown underneath the main content. */
  description?: string,
  /** Drag and drop props provided by react-beautiful-dnd. Please do not use
   * this unless using react-beautiful-dnd */
  dnd?: DnDType,
  /** Content to be shown after the main content. Shown to the right of content (or to the left
   * in RTL mode). */
  elemAfter?: Node,
  /** Content to be shown before the main content. Shown to the left of content (or to the right
   * in RTL mode). */
  elemBefore?: Node,
  /** Link that the user will be redirected to when the item is clicked. If omitted, a
   *  non-hyperlink component will be rendered. */
  href?: string,
  /** Causes the item to be rendered with reduced spacing. */
  isCompact?: boolean,
  /** Causes the item to appear in a disabled state and click behaviours will not be triggered. */
  isDisabled?: boolean,
  /** Used to apply correct dragging styles when also using react-beautiful-dnd. */
  isDragging?: boolean,
  /** Causes the item to still be rendered, but with `display: none` applied. */
  isHidden?: boolean,
  /** Causes the item to appear with a persistent selected background state. */
  isSelected?: boolean,
  /** Function to be called when the item is clicked, Receives the MouseEvent. */
  onClick?: Function,
  /** Function to be called when the item is pressed with a keyboard,
   * Receives the KeyboardEvent. */
  onKeyDown?: Function,
  /** Standard onmouseenter event */
  onMouseEnter?: Function,
  /** Standard onmouseleave event */
  onMouseLeave?: Function,
  /** Allows the role attribute of the item to be altered from it's default of
   *  `role="button"` */
  role?: string,
  /** Target frame for item `href` link to be aimed at. */
  target?: string,
  /** Standard browser title to be displayed on the item when hovered. */
  title?: string,
};

/**
  Type of keyboard event that triggers which key will should close the drawer.
*/
export type CloseTrigger = 'backButton' | 'blanket' | 'escKey';
