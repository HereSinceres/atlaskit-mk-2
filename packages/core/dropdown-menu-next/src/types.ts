import { UIAnalyticsEvent } from '@atlaskit/analytics-next';
import { Placement } from '@atlaskit/popper';

export type DefaultButtonProps = {
  text: string;
  onClick?: (
    e: React.MouseEvent<HTMLElement>,
    analyticsEvent: UIAnalyticsEvent,
  ) => void;
};

export type DropdownMenuProps = {
  /**
   * Controls the appearance of the menu.
   * Default menu has scroll after its height exceeds the pre-defined amount.
   * Tall menu has no scroll until the height exceeds the height of the viewport.
   */
  appearance?: 'default' | 'tall';
  /** Value passed to the Layer component to determine when to reposition the droplist */
  boundariesElement?: 'viewport' | 'window' | 'scrollParent';
  /** Default menu open state */
  defaultIsOpen?: boolean;
  /** HTML Id for testing etc */
  id?: string;
  /** Position of the menu. See the documentation of @atlaskit/layer for more details. */
  position?: Placement;
  /** Option to fit dropdown menu width to its parent width */
  shouldFitContainer?: boolean;
  /** Allows the dropdown menu to be placed on the opposite side of its trigger if it does not
   * fit in the viewport. */
  shouldFlip?: boolean;
  /**testId maps to data-test-id for testing in your application */
  testId?: string;
};

export type DropdownState = {
  appearance: 'default' | 'tall';
  boundriesElement: 'viewport' | 'window' | 'scrollParent';
  isOpen: boolean;
  position: Placement;
  shouldFitContainer: boolean;
  shouldFlip: boolean;
  selectionIndex: number;
};

export type DropdownTriggerProps = {
  text?: string;
};
