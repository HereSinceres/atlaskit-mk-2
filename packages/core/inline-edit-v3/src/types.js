// @flow
import { type Node } from 'react';

type Props = {
  /** Label above the input */
  label?: string,
  /** Component to be shown when not in edit view */
  readView: Node,
  /** Component to be shown when editing. Should be an Atlaskit input. */
  editView: Node,
  /** Whether the component shows the readView or the editView. */
  isEditing: boolean,
  /** The initial value of the inline edit. */
  defaultValue: any,
  /** Validation function handled by final-form. */
  validate?: (
    value: any,
    formState: Object,
    fieldState: Object,
  ) => string | void | Promise<string | void>,

  /** Contained in defaultProps */
  /** Handler called when readView is clicked. */
  onEditRequested: any => mixed,
  /** Handler called editView is closed and changes are confirmed.
   * Field value is passed as an argument to this function. */
  onConfirm: any => mixed,
  /** Handler called when checkmark is. */
  onCancel: any => mixed,
  /** Set whether onConfirm should be called on blur.*/
  disableConfirmOnBlur: boolean,
  /** Sets whether the checkmark and cross are displayed in the bottom right of the field. */
  hideActionButtons: boolean,
  /** The text announced to screen readers when focusing on the edit button. */
  editButtonLabel: string,
  /** The text announced to screen readers when focusing on the confirm button. */
  confirmButtonLabel: string,
  /** The text announced to screen readers when focusing on the cancel button. */
  cancelButtonLabel: string,
};

type DefaultProps = {
  /** Handler called when readView is clicked. */
  onEditRequested: any => mixed,
  /** Handler called editView is closed and changes are confirmed.
   * Field value is passed as an argument to this function. */
  onConfirm: any => mixed,
  /** Handler called when checkmark is. */
  onCancel: any => mixed,
  /** Set whether onConfirm should be called on blur.*/
  disableConfirmOnBlur: boolean,
  /** Sets whether the checkmark and cross are displayed in the bottom right of the field. */
  hideActionButtons: boolean,
  /** The text announced to screen readers when focusing on the edit button. */
  editButtonLabel: string,
  /** The text announced to screen readers when focusing on the confirm button. */
  confirmButtonLabel: string,
  /** The text announced to screen readers when focusing on the cancel button. */
  cancelButtonLabel: string,
};
