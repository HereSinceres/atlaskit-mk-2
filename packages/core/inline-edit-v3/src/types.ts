import * as React from 'react';

export interface Props {
  /** Label above the input */
  label?: string;
  /** Component to be shown when not in edit view */
  readView: React.ReactChild;
  /** Component to be shown when editing. Should be an Atlaskit input. */
  editView: (fieldProps: FieldProps, isInvalid?: boolean) => React.ReactChild;
  /** Whether the component shows the readView or the editView. */
  isEditing: boolean;
  /** The initial value of the inline edit. */
  defaultValue: any;
  /** Validation function handled by final-form. */
  validate?: (
    value: any,
    formState: {},
    fieldState: {},
  ) => string | void | Promise<string | void>;
  /** Handler called when readView is clicked. */
  onEditRequested: () => void;
  /**
   * Handler called editView is closed and changes are confirmed.
   * Field value is passed as an argument to this function.
   */
  onConfirm: (value: any) => void;
  /** Handler called when checkmark is. */
  onCancel: () => void;

  /** Contained in defaultProps */
  /** Set whether onConfirm should be called on blur. */
  disableConfirmOnBlur?: boolean;
  /** Sets whether the checkmark and cross are displayed in the bottom right of the field. */
  hideActionButtons?: boolean;
  /** The text announced to screen readers when focusing on the edit button. */
  editButtonLabel?: string;
  /** The text announced to screen readers when focusing on the confirm button. */
  confirmButtonLabel?: string;
  /** The text announced to screen readers when focusing on the cancel button. */
  cancelButtonLabel?: string;
  /** Determines whether the read view has 100% width within its container, or whether it fits the content */
  readViewFitContainerWidth?: boolean;
}

export type RenderChildrenProps = {
  /** Label above the input */
  label?: string;
  /** Whether the component shows the readView or the editView. */
  isEditing: boolean;
  /** The initial value of the inline edit. */
  defaultValue: any;
  /** Validation function handled by final-form. */
  validate?: (
    value: any,
    formState: {},
    fieldState: {},
  ) => string | void | Promise<string | void>;
  /** Handler called when readView is clicked. */
  onEditRequested: () => void;
  /**
   * Handler called editView is closed and changes are confirmed.
   * Field value is passed as an argument to this function.
   */
  onConfirm: (value: any) => void;
  /** Handler called when checkmark is. */
  onCancel: () => void;
  /** Elements to render inside InlineEdit */
  children: (isEditing: boolean, fieldProps?: FieldProps) => React.ReactChild;

  /** Contained in defaultProps */
  /** Set whether onConfirm should be called on blur. */
  disableConfirmOnBlur?: boolean;
  /** Sets whether the checkmark and cross are displayed in the bottom right of the field. */
  hideActionButtons?: boolean;
  /** The text announced to screen readers when focusing on the edit button. */
  editButtonLabel?: string;
  /** The text announced to screen readers when focusing on the confirm button. */
  confirmButtonLabel?: string;
  /** The text announced to screen readers when focusing on the cancel button. */
  cancelButtonLabel?: string;
};

/** These interfaces will be exported from the Form package once it is converted to Typescript */
export interface FormProps {
  onSubmit: (e: React.FormEvent) => void;
  ref: React.RefObject<HTMLFormElement>;
  onKeyDown: (e: React.KeyboardEvent) => void;
}

export interface FormChildProps {
  formProps: FormProps;
  dirty: boolean;
  submitting: boolean;
  disabled: boolean;
  getValues: () => {};
}

export interface FieldProps {
  id: string;
  isRequired: boolean;
  isDisabled: boolean;
  isInvalid: boolean;
  onChange: (e: any) => any;
  onBlur: () => any;
  onFocus: () => any;
  value: any;
  'aria-invalid': 'true' | 'false';
  'aria-labelledby': string;
}

export interface Meta {
  dirty: boolean;
  touched: boolean;
  valid: boolean;
  error: any;
  submitError: any;
}

export interface FieldChildProps {
  fieldProps: FieldProps;
  error: any;
  meta: Meta;
}

/** This interface will be exported once Inline Dialog is converted to Typescript */
export type Placement =
  | 'auto-start'
  | 'auto'
  | 'auto-end'
  | 'top-start'
  | 'top'
  | 'top-end'
  | 'right-start'
  | 'right'
  | 'right-end'
  | 'bottom-end'
  | 'bottom'
  | 'bottom-start'
  | 'left-end'
  | 'left'
  | 'left-start';

export interface InlineDialogProps {
  children?: React.ReactChild;
  content?: React.ReactChild;
  isOpen?: boolean;
  onContentBlur?: () => void;
  onContentClick?: () => void;
  onContentFocus?: () => void;
  onClose?: (e: { isOpen: false; event: any }) => void;
  placement?: Placement;
}
