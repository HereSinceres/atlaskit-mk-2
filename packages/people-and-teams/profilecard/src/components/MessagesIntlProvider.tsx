import * as React from 'react';
import { IntlProvider, injectIntl } from 'react-intl';
import { getMessagesForLocale } from '../internal/i18n-util';
import { MessageIntlProviderProps } from '../types';

class MessagesIntlProvider extends React.PureComponent<
  MessageIntlProviderProps
> {
  render() {
    const { intl, children } = this.props;

    return (
      <IntlProvider messages={getMessagesForLocale(intl.locale)}>
        {children}
      </IntlProvider>
    );
  }
}

export default injectIntl(MessagesIntlProvider);
