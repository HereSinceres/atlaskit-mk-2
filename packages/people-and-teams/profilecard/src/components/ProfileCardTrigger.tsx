import React from 'react';
import { Popper } from '@atlaskit/popper';
import Portal from '@atlaskit/portal';
import { layers } from '@atlaskit/theme';
import NodeResolver from 'react-node-resolver';

import LoadingState from './LoadingState';
import ErrorMessage from './ErrorMessage';
import Profilecard from './ProfileCard';
import withOuterListeners from './withOuterListeners';
import filterActions from '../internal/filterActions';

import { CardElevationWrapper, CardTriggerWrapper } from '../styled/Card';

const CardElevationWrapperWithOuter = withOuterListeners(CardElevationWrapper);

import {
  ProfileCardTriggerProps,
  ProfileCardTriggerState,
  ProfileCardAction,
  ProfilecardProps,
  ProfileCardClientData,
} from '../types';

class ProfilecardTrigger extends React.Component<
  ProfileCardTriggerProps,
  ProfileCardTriggerState
> {
  static defaultProps: Partial<ProfileCardTriggerProps> = {
    actions: [],
    trigger: 'hover',
    customElevation: 'e200',
  };

  _isMounted: boolean;
  wrapperRef: HTMLElement | null;
  targetRef: HTMLElement | null;

  showDelay: number = 500;
  hideDelay: number = 500;
  showTimer: any;
  hideTimer: any;

  constructor(props: ProfileCardTriggerProps) {
    super(props);
    this._isMounted = false;

    // In constrast to hover, click interactions should be instantaneous
    if (this.props.trigger === 'click') {
      this.showDelay = 0;
      this.hideDelay = 0;
    }
  }

  state: ProfileCardTriggerState = {
    visible: false,
    isLoading: false,
    hasError: false,
    error: null,
    data: null,
  };

  componentDidMount() {
    this._isMounted = true;
  }

  // @FIXME do we need this for the TRIGGER component?
  componentDidUpdate(prevProps: ProfileCardTriggerProps) {
    const { userId, cloudId } = this.props;
    if (userId !== prevProps.userId || cloudId !== prevProps.cloudId) {
      this.clientFetchProfile();
    }
  }

  componentWillUnmount() {
    this._isMounted = false;
  }

  clientFetchProfile = () => {
    const { cloudId, userId } = this.props;

    this.setState({
      isLoading: true,
      hasError: false,
      data: null,
    });

    this.props.resourceClient
      .getProfile(cloudId, userId)
      .then(
        res => this.handleClientSuccess(res),
        err => this.handleClientError(err),
      )
      .catch(err => this.handleClientError(err));
  };

  handleClientSuccess(res: ProfileCardClientData) {
    if (!this._isMounted) {
      return;
    }

    this.setState({
      isLoading: false,
      hasError: false,
      data: res,
    });
  }

  handleClientError(err: any) {
    if (!this._isMounted) {
      return;
    }
    this.setState({
      isLoading: false,
      hasError: true,
      error: err,
    });
  }

  filterActions = (): ProfileCardAction[] =>
    filterActions(this.props.actions, this.state.data);

  hideProfilecard = () => {
    clearTimeout(this.showTimer);

    this.hideTimer = setTimeout(() => {
      this.setState({ visible: false });
    }, this.hideDelay);
  };

  showProfilecard = () => {
    if (!this.state.visible) {
      this.clientFetchProfile();
    }

    clearTimeout(this.hideTimer);

    this.showTimer = setTimeout(() => {
      this.setState({ visible: true });
    }, this.showDelay);
  };

  getContainerListeners = () => {
    const containerListeners = {} as any;

    if (this.props.trigger === 'hover') {
      containerListeners.onMouseEnter = this.showProfilecard;
      containerListeners.onMouseLeave = this.hideProfilecard;
    } else {
      containerListeners.onClick = this.showProfilecard;
    }

    return containerListeners;
  };

  getLayerListeners = () => {
    const layerListeners = {} as any;

    if (this.props.trigger !== 'hover') {
      layerListeners.handleClickOutside = this.hideProfilecard;
      layerListeners.handleEscapeKeydown = this.hideProfilecard;
    }

    return layerListeners;
  };

  renderProfileCard = () => {
    const newProps: ProfilecardProps = {
      clientFetchProfile: this.clientFetchProfile,
      analytics: this.props.analytics,
      ...this.state.data,
    };

    // @FIXME figure out better way to display error state
    // maybe putting it back into ProfileCard?!
    // put it back!!!
    if (this.state.hasError) {
      return (
        <CardElevationWrapper customElevation="none">
          <ErrorMessage
            errorType={this.state.error}
            reload={this.clientFetchProfile}
          />
        </CardElevationWrapper>
      );
    }

    return (
      <Profilecard
        {...newProps}
        actions={this.filterActions()}
        customElevation="none"
      />
    );
  };

  renderWithPopper = (element: React.ReactNode) => {
    const WrapperElement =
      this.props.trigger === 'hover'
        ? CardElevationWrapper
        : CardElevationWrapperWithOuter;

    return (
      <Popper referenceElement={this.targetRef} placement={this.props.position}>
        {({ ref, style }) => (
          <WrapperElement
            style={style}
            innerRef={ref}
            {...this.getContainerListeners()}
            {...this.getLayerListeners()}
            customElevation={this.props.customElevation}
          >
            {element}
          </WrapperElement>
        )}
      </Popper>
    );
  };

  renderLoading = () =>
    this.state.visible && this.state.isLoading && this.targetRef
      ? this.renderWithPopper(<LoadingState />)
      : null;

  renderProfileCardLoaded = () =>
    this.state.visible && !this.state.isLoading && this.targetRef
      ? this.renderWithPopper(this.renderProfileCard())
      : null;

  renderWithTrigger = () => (
    <>
      <CardTriggerWrapper
        {...this.getContainerListeners()}
        ref={wrapperRef => {
          this.wrapperRef = wrapperRef;
        }}
      >
        <NodeResolver
          innerRef={targetRef => {
            this.targetRef = targetRef;
          }}
        >
          {this.props.children}
        </NodeResolver>
      </CardTriggerWrapper>
      <Portal zIndex={layers.tooltip()}>
        {this.renderLoading()}
        {this.renderProfileCardLoaded()}
      </Portal>
    </>
  );

  render() {
    if (this.props.children) {
      return this.renderWithTrigger();
    } else {
      console.error('ProfileCardTrigger must have props.children');
      return null;
    }
  }
}

export default ProfilecardTrigger;
