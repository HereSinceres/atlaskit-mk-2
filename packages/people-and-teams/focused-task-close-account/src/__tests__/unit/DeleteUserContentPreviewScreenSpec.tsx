import * as React from 'react';
import { shallow } from 'enzyme';
import { DeleteUserContentPreviewScreen } from '../../components/DeleteUserContentPreviewScreen';
import { catherineHirons } from '../../mocks/users';
import { DeleteUserContentPreviewScreenProps } from '../../components/DeleteUserContentPreviewScreen/types';

const defaultProps: DeleteUserContentPreviewScreenProps = {
  isCurrentUser: false,
  user: catherineHirons,
  preferenceSelection: (name: string) => jest.fn(),
};

const render = (props = {}) =>
  shallow(<DeleteUserContentPreviewScreen {...defaultProps} {...props} />);

test('DeleteUserContentPreviewScreen', () => {
  expect(render()).toMatchSnapshot();
});

describe('selectAdminOrSelfCopy', () => {
  test('selects admin copy if delete candidate is not current user', () => {
    const selectAdminOrSelfCopy = (render({
      isCurrentUser: false,
    }).instance() as DeleteUserContentPreviewScreen).selectAdminOrSelfCopy;
    expect(selectAdminOrSelfCopy('admin' as any, 'self' as any)).toBe('admin');
  });

  test('selects self copy if delete candidate is current user', () => {
    const selectAdminOrSelfCopy = (render({
      isCurrentUser: true,
    }).instance() as DeleteUserContentPreviewScreen).selectAdminOrSelfCopy;
    expect(selectAdminOrSelfCopy('admin' as any, 'self' as any)).toBe('self');
  });
});

describe('handleClickSelection', () => {
  test('changes the isSelected parameter of the element', () => {
    const wrapper = render();
    const divWrapper = wrapper.find('.nameSectionCard');
    shallow(<div>{divWrapper}</div>)
      .childAt(0)
      .simulate('click');
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });
});
