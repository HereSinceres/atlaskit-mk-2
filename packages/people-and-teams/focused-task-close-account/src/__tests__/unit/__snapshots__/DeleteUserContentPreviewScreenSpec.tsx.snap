// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteUserContentPreviewScreen 1`] = `
<InjectIntl(MessagesIntlProvider)>
  <styled.div>
    <styled.div>
      <FormattedMessage
        defaultMessage="Tell us how do you want the user to be remembered"
        description="Heading for the screen that explains admin would prefer user to be remembered"
        id="focused-task-close-account.delete-account.content-preview.heading.admin"
        values={Object {}}
      />
    </styled.div>
    <FormattedMessage
      defaultMessage="After the user’s account is deleted, they’ll appear as “Former user” to other users. Please take a moment to answer our survey question."
      description="First paragraph for the survey for admins"
      id="focused-task-close-account.delete-account.content-preview.paragraph.survey.admin"
      tagName="p"
      values={Object {}}
    />
    <FormattedMessage
      defaultMessage="If we gave you a choice, how would you like the deleted user to appear to other users?"
      description="Second paragraph for the survey for admins"
      id="focused-task-close-account.delete-account.content-preview.line.survey.admin"
      tagName="p"
      values={Object {}}
    />
    <styled.div
      className="nameSectionCard"
      isSelected={false}
      onClick={[Function]}
    >
      <styled.div>
        <WithAnalyticsContext(WithAnalyticsEvents(mapProps(withPseudoState(Avatar))))
          size="large"
          src=""
        />
      </styled.div>
      <styled.div>
        Catherine Hirons
      </styled.div>
    </styled.div>
    <styled.div
      className="formerUserSectionCard"
      isSelected={false}
      onClick={[Function]}
    >
      <styled.div>
        <WithAnalyticsContext(WithAnalyticsEvents(mapProps(withPseudoState(Avatar))))
          size="large"
          src=""
        />
      </styled.div>
      <styled.div>
        Former User
      </styled.div>
    </styled.div>
    <FormattedHTMLMessage
      defaultMessage="Note: Answering this survey will help us make a better experience for all users. The user will still appear as “Former user” after you delete their account."
      description="Note to admins that the users would still be saved as Former User"
      id="focused-task-close-account.delete-account.content-preview.footnote.admin"
      tagName="p"
      values={Object {}}
    />
  </styled.div>
</InjectIntl(MessagesIntlProvider)>
`;

exports[`handleClickSelection changes the isSelected parameter of the element 1`] = `
<InjectIntl(MessagesIntlProvider)>
  <styled.div>
    <styled.div>
      <FormattedMessage
        defaultMessage="Tell us how do you want the user to be remembered"
        description="Heading for the screen that explains admin would prefer user to be remembered"
        id="focused-task-close-account.delete-account.content-preview.heading.admin"
        values={Object {}}
      />
    </styled.div>
    <FormattedMessage
      defaultMessage="After the user’s account is deleted, they’ll appear as “Former user” to other users. Please take a moment to answer our survey question."
      description="First paragraph for the survey for admins"
      id="focused-task-close-account.delete-account.content-preview.paragraph.survey.admin"
      tagName="p"
      values={Object {}}
    />
    <FormattedMessage
      defaultMessage="If we gave you a choice, how would you like the deleted user to appear to other users?"
      description="Second paragraph for the survey for admins"
      id="focused-task-close-account.delete-account.content-preview.line.survey.admin"
      tagName="p"
      values={Object {}}
    />
    <styled.div
      className="nameSectionCard"
      isSelected={true}
      onClick={[Function]}
    >
      <styled.div>
        <WithAnalyticsContext(WithAnalyticsEvents(mapProps(withPseudoState(Avatar))))
          size="large"
          src=""
        />
      </styled.div>
      <styled.div>
        Catherine Hirons
      </styled.div>
    </styled.div>
    <styled.div
      className="formerUserSectionCard"
      isSelected={false}
      onClick={[Function]}
    >
      <styled.div>
        <WithAnalyticsContext(WithAnalyticsEvents(mapProps(withPseudoState(Avatar))))
          size="large"
          src=""
        />
      </styled.div>
      <styled.div>
        Former User
      </styled.div>
    </styled.div>
    <FormattedHTMLMessage
      defaultMessage="Note: Answering this survey will help us make a better experience for all users. The user will still appear as “Former user” after you delete their account."
      description="Note to admins that the users would still be saved as Former User"
      id="focused-task-close-account.delete-account.content-preview.footnote.admin"
      tagName="p"
      values={Object {}}
    />
  </styled.div>
</InjectIntl(MessagesIntlProvider)>
`;
